lerobot/scripts/server/gym_manipulator.py

```diff
@@ -0,0 +1,1377 @@
+                ee_action_space_params.y_step_size,
+                ee_action_space_params.z_step_size,
+            ]
+        )
+        if self.use_gripper:
+            action_space_bounds = np.concatenate([action_space_bounds, [1.0]])
+        ee_action_space = gym.spaces.Box(
+            low=-action_space_bounds,
+            high=action_space_bounds,
+            shape=(3 + int(self.use_gripper),),
+            dtype=np.float32,
+        )
+        if isinstance(self.action_space, gym.spaces.Tuple):
+            self.action_space = gym.spaces.Tuple((ee_action_space, self.action_space[1]))
+        else:
+            self.action_space = ee_action_space
+
+        self.bounds = ee_action_space_params.bounds
+
+    def action(self, action):
+        is_intervention = False
+        desired_ee_pos = np.eye(4)
+        if isinstance(action, tuple):
+            action, _ = action
+
+        if self.use_gripper:
+            gripper_command = action[-1]
+            action = action[:-1]
+
+        current_joint_pos = self.unwrapped.robot.follower_arms["main"].read("Present_Position")
+        current_ee_pos = self.fk_function(current_joint_pos)
+        if isinstance(action, torch.Tensor):
+            action = action.cpu().numpy()
+        desired_ee_pos[:3, 3] = np.clip(
+            current_ee_pos[:3, 3] + action,
+            self.bounds["min"],
+            self.bounds["max"],
+        )
+        target_joint_pos = self.kinematics.ik(
+            current_joint_pos,
+            desired_ee_pos,
+            position_only=True,
+            fk_func=self.fk_function,
+        )
+        if self.use_gripper:
+            # Quantize gripper command to -1, 0 or 1
+            if gripper_command < -0.2:
+                gripper_command = -1.0
+            elif gripper_command > 0.2:
+                gripper_command = 1.0
+            else:
+                gripper_command = 0.0
+
+            gripper_state = self.unwrapped.robot.follower_arms["main"].read("Present_Position")[-1]
+            gripper_action = np.clip(gripper_state + gripper_command, 0, MAX_GRIPPER_COMMAND)
+            target_joint_pos[-1] = gripper_action
+
+        return target_joint_pos, is_intervention
+
+
+class EEObservationWrapper(gym.ObservationWrapper):
+    def __init__(self, env, ee_pose_limits):
+        super().__init__(env)
+
+        # Extend observation space to include end effector pose
+        prev_space = self.observation_space["observation.state"]
+
+        self.observation_space["observation.state"] = gym.spaces.Box(
+            low=np.concatenate([prev_space.low, ee_pose_limits["min"]]),
+            high=np.concatenate([prev_space.high, ee_pose_limits["max"]]),
+            shape=(prev_space.shape[0] + 3,),
+            dtype=np.float32,
+        )
+
+        # Initialize kinematics instance for the appropriate robot type
+        robot_type = getattr(env.unwrapped.robot.config, "robot_type", "so100")
```

Comment:
```suggestion
        robot_type = getattr(env.unwrapped.robot.config, "type", "so100")
```
Also I'm a bit against the getattr with default so100 as it leads to silent bugs. Took me a minute to realize why my moss was constrained